src/main/kotlin/com/exactpro/th2/kafka/client/KafkaClientsFactory.kt

```diff
@@ -31,27 +33,43 @@ import java.util.Properties
 class KafkaClientsFactory(private val config: Config) {
     fun getKafkaConsumer(): Consumer<String, ByteArray> = KafkaConsumer(
         Properties().apply {
-            putAll(mapOf(
+            val props = mutableMapOf(
                 ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG to config.bootstrapServers,
                 ConsumerConfig.GROUP_ID_CONFIG to config.groupId,
                 ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG to false,
                 ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG to StringDeserializer::class.java,
                 ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG to ByteArrayDeserializer::class.java,
                 ConsumerConfig.RECONNECT_BACKOFF_MS_CONFIG to config.reconnectBackoffMs,
-                ConsumerConfig.RECONNECT_BACKOFF_MAX_MS_CONFIG to config.reconnectBackoffMaxMs
-            ))
+                ConsumerConfig.RECONNECT_BACKOFF_MAX_MS_CONFIG to config.reconnectBackoffMaxMs,
+
+                CommonClientConfigs.SECURITY_PROTOCOL_CONFIG to config.kafkaSecurityProtocol,
+                SaslConfigs.SASL_KERBEROS_SERVICE_NAME to config.kafkaSaslKerberosServiceName,
+                SaslConfigs.SASL_MECHANISM to config.kafkaSaslMechanism,
+                SaslConfigs.SASL_JAAS_CONFIG to config.kafkaSaslJaasConfig
+            )
```

Comment:
Maybe we should use `buildMap` instead and add SASL parameters only if they are set.

Also, I think we need validation for those properties (which of them is required for a secure connection?)